src/Razor/test/RazorLanguage.Test/TestFiles/IntegrationTests/ComponentDesignTimeCodeGenerationTest/ChildComponent_NonGenericParameterizedChildContent_TypeInference/TestComponent.codegen.cs

```diff
@@ -0,0 +1,76 @@
+// <auto-generated/>
+#pragma warning disable 1591
+namespace Test
+{
+    #line hidden
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Microsoft.AspNetCore.Components;
+    public class TestComponent : Microsoft.AspNetCore.Components.ComponentBase
+    {
+        #pragma warning disable 219
+        private void __RazorDirectiveTokenHelpers__() {
+        }
+        #pragma warning restore 219
+        #pragma warning disable 0414
+        private static System.Object __o = null;
+        #pragma warning restore 0414
+        #pragma warning disable 1998
+        protected override void BuildRenderTree(Microsoft.AspNetCore.Components.RenderTree.RenderTreeBuilder builder)
+        {
+            __Blazor.Test.TestComponent.TypeInference.CreateMyComponent_0(builder, -1, -1, 
+#nullable restore
+#line 1 "x:\dir\subdir\Test\TestComponent.cshtml"
+                     "hi"
+
+#line default
+#line hidden
+#nullable disable
+            , -1, (context) => (builder2) => {
+#nullable restore
+#line 2 "x:\dir\subdir\Test\TestComponent.cshtml"
+              __o = context.ToLower();
+
+#line default
+#line hidden
+#nullable disable
+            }
+            , -1, (context) => (builder2) => {
+#nullable restore
+#line 3 "x:\dir\subdir\Test\TestComponent.cshtml"
+          __o = context;
+
+#line default
+#line hidden
+#nullable disable
+            }
+            );
+#nullable restore
+#line 1 "x:\dir\subdir\Test\TestComponent.cshtml"
+__o = typeof(MyComponent<>);
+
+#line default
+#line hidden
+#nullable disable
+        }
+        #pragma warning restore 1998
+    }
+}
+namespace __Blazor.Test.TestComponent
+{
+    #line hidden
+    internal static class TypeInference
+    {
+        public static void CreateMyComponent_0<TItem>(global::Microsoft.AspNetCore.Components.RenderTree.RenderTreeBuilder builder, int seq, int __seq0, TItem __arg0, int __seq1, global::Microsoft.AspNetCore.Components.RenderFragment<TItem> __arg1, int __seq2, global::Microsoft.AspNetCore.Components.RenderFragment<System.Int32> __arg2)
```

Comment:
The change above makes a difference in this line.
Before:
`global::Microsoft.AspNetCore.Components.RenderFragment<System.Object> __arg2`

After
`global::Microsoft.AspNetCore.Components.RenderFragment<System.Int32> __arg2`